sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-full-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -1484,6 +1486,495 @@ describe(
+          const uploadedFile = getUploadedFile(sftpClientMock);
+          const uploadedFileName = getUploadedFileName();
+          expect(uploadedFile.remoteFilePath).toBe(uploadedFileName);
+          // One record should be sent.
+          expect(result).toStrictEqual([
+            `Generated file: ${uploadedFileName}`,
+            "Uploaded records: 1",
+          ]);
+          expect(
+            mockedJob.containLogMessages([
+              `All calculations were saved and disbursement was set to '${DisbursementScheduleStatus.ReadyToSend}'.`,
+              "Checking offering for aviation credential types to add a restriction.",
+              `New restriction ${RestrictionCode.AVCP} for the aviation credential type ${aviationCredentialType} was added.`,
+            ]),
+          ).toBe(true);
+          const [disbursement] =
+            aviationCredentialApplication.currentAssessment
+              .disbursementSchedules;
+          // Validate the disbursement schedule status and existing and new student restriction.
+          const updatedDisbursement =
+            await loadDisbursementAndStudentRestrictions(db, disbursement.id);
+          expect(updatedDisbursement).toEqual({
+            id: disbursement.id,
+            disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+            studentAssessment: {
+              id: aviationCredentialApplication.currentAssessment.id,
+              application: {
+                id: aviationCredentialApplication.id,
+                student: {
+                  id: aviationCredentialApplication.student.id,
+                  // Student should have 2 student restrictions where one is added from previous funding
+                  // and the other one is added from the current e-cert generation.
+                  studentRestrictions: [
+                    {
+                      id: differentAviationCredentialRestriction.id,
+                      isActive: true,
+                      restriction: {
+                        id: restriction.id,
+                        restrictionCode:
+                          differentAviationCredentialRestrictionCode,
+                      },
+                    },
+                    {
+                      id: expect.any(Number),
+                      isActive: true,
+                      restriction: {
+                        id: expect.any(Number),
+                        restrictionCode: RestrictionCode.AVCP,
+                      },
+                    },
+                  ],
+                },
+              },
+            },
+          });
+        },
+      );
+
+      it(
+        "Should generate e-cert and not block the disbursement for aviation credential type 'commercialPilotTraining'" +
+          ` when the student already has an active restriction ${RestrictionCode.AVCP} added due to previous funding of the same aviation credential type` +
+          " but the restriction was bypassed.",
```

Comment:
Is there a specific scenario to be tested for aviation and bypasses that was not already tested by the existing bypass scenarios?

Reply:
To test the scenario that 

-> if a aviation restriction for the credential type already exist, it must create another restriction

Bypass was just the way to reach the step.